### ISSUE TITLE
WebHooks for ASP.NET Core

### ISSUE THREAD

**Issue by smbecker** [aspnet/WebHooks#5]
Is there a plan to support ASP.NET 5?

**Comment by Eilon** [comment]
Of course :smile:

**Comment by garora** [comment]
We have ASP.NET5 Beta7 released, so, what we are meant with the support of ASP.NET5?
I believe @HenrikFrystykNielsen - already planned :+1:

**Comment by Eilon** [comment]
There is no support yet for Web Hooks on ASP.NET 5. It currently works with ASP.NET 4.x only.

**Comment by garora** [comment]
@Eilon thanks for confirming but as per your above reply, I understood that its in plan?

**Comment by Eilon** [comment]
We intend to support it. There is no specific date.

**Comment by nicofr83** [comment]
Any news/date on this support ? I need to run a web hook under a linux box and I would hate to return to javascript/node... Thanks

**Comment by Eilon** [comment]
Sorry no updates at this time.

**Comment by rposener** [comment]
Just FYI - I have a pending pull request with mostly working port of the base and custom sender/receiver.  Hopefully we will see this soon.  Will need some help building up the supported receivers for the various supported hooks out of the box - if you want to help contribute those once the base stuff is sync'd in.

**Comment by Eilon** [comment]
@rposener I do intend to take a look at it, but I just haven't yet found the time. It's kind of a big PR :smile:

**Comment by mikeesouth** [comment]
Any news? I am also intrested in this.

**Comment by rposener** [comment]
Still waiting to hear from the asp.net core team.  The code base has moved a lot since this pull request.  Anyone interested is just writing out own since MS is being so slow?  New tooling should be out this week helping us move onto CLI.  Unless MS give confirmation that they will actually review and give input on a pull request to get this done.  Kind of frustrating they are not embracing OSS community any more than this on this project!

**Comment by HenrikFrystykNielsen** [comment]
I know this is frustrating and I apologize. With RC2 being out, my hope is that we can get some more cycles from the right people. Stay tuned!

**Comment by mikeesouth** [comment]
Can we help out somehow? I'm really waiting for this so being able to help our getting an ETA would be great :)

**Comment by rposener** [comment]
The pull request is there, updated to rc2 with custom sender and receiver working in sample.

Might be good if you at minimum grabbed the source from pull source https://github.com/rposener/webhooks and gave it a test and provide comments. 

Just need some review from asp.net team, and it to be merged in.  If you want to fork a copy of the pull request and build xunit tests to convert the existing test sets.  Then we just need to crank through the other receivers and samples.

**Comment by mikeesouth** [comment]
Unfortunately, I'm not much help here. I'm having troubles understanding your changes @rposener. The commits have thousands of changed files and I'm too new to loading custom libraries so I'm afraid I'm not much help here. I'll need some serious instructions on how to incorporate a custom built lib into an ASP.NET Core project. I would also like to use the Slack receiver but I assume I can mimic it with a custom receiver.
So, any help to get me on track or even better - an official merge into the core library would be great :)

**Comment by ups216** [comment]
hi, asp.net core 1.0 is now RTM. Is there any progress on porting this to core?

**Comment by bragma** [comment]
I'd really like to see this ported to Core too. Please do it! :)

**Comment by ArturKarbone** [comment]
@rposener,  I just pulled your fork. Was not able to run it, since it's still requires RC2. Anyway I looked through the code. At the first glance it looks there are no breaking changes in terms of method's signatures/contracts. 
 
However, there is something Core specific stuff, like initialization extensions, middleWare, etc.


I mean to support Core, WebHooks should support two code bases now, right? Otherwise previous package clients will be in trouble. Just wandering what is the best way to go here?

**Comment by tedvanderveen** [comment]
@Eilon as you assigned this issue to yourself, can you please update on status? With Core 1.0.0 released it would be a good time to look at the PR that @rposener invested time in developing?

**Comment by coderabsolute** [comment]
Any timelines to have the Webhook functionality released?

**Comment by BenHayat** [comment]
Is WH based on similar principals as SignalR? Do they overlap or complement each other?

Can't wait for these two to come to ASP.Net Core.

Thanks!
..Ben

**Comment by consigliory** [comment]
+1

**Comment by divad4686** [comment]
+1

**Comment by SimoneBWS** [comment]
👍

**Comment by hasan-assi** [comment]
+1

**Comment by toast-gear** [comment]
+1

**Comment by Beerd-FR** [comment]
Any news about asp.net core support ?

**Comment by stodolos** [comment]
Any updates on Core support timeline?

**Comment by SimonLuckenuik** [comment]
@HenrikFrystykNielsen is this still on the roadmap?

**Comment by xstof** [comment]
+1 Any updates would be welcome.

**Comment by SimonLuckenuik** [comment]
Worst case just the sender side, it would help creating dotnet core apps to submit messages and have the webhooks publisher in classic .NET...

**Comment by divad4686** [comment]
I don't think anyone is maintaining this anymore.

**Comment by HenrikFrystykNielsen** [comment]
I apologize for having gone quiet -- I have completely blocked by other work. Will get back to it soon!

**Comment by Sky4CE** [comment]
Guys, is it possible to use webhooks somehow in aspnet core app before aspnet team implement this? I'm interesting in receiving webhook events from stripe, maybe somebody know some solution? Because at this point as I understand it is absolutely impossible to use stripe framework at all if you need to receive events.

**Comment by garora** [comment]
For porting .Net Core support we should first see this [#2022](https://github.com/aspnet/Home/issues/2022)

**Comment by cwoolum** [comment]
Looks like #2202 has basically been been resolved with aspnet core 2.0-preview2. Now that standard .Net libraries will be supported, what would it take to add webhooks support? @rposener , Is your PR still valid?

**Comment by Eilon** [comment]
Hi everyone, just a heads up that we're getting ready to work on this - finally! Sorry for the long delay, we needed to finish up ASP.NET Core 2.0, which is almost out the door.

We're planning for the next release of ASP.NET Web Hooks to be supported in ASP.NET Core. We'll share more details as we figure out what exactly it should look like.

**Comment by rposener** [comment]
Excellent - have not looked my original work in a long time (maybe a year?) In any event - have a feeling such has moved it needs a completely fresh review / look.  We went in another direction, but if there is anything I can do to help, just ask. We will be looking to pull this into our projects as soon as it's available.

**Comment by Eilon** [comment]
@rposener thanks! The amount of requests we're getting for Web Hooks on ASP.NET Core is only increasing, so we hope to get something out as soon as possible.

**Comment by garora** [comment]
@Eilon - Great thanks for this decision and the news you shared above. There is much awaited things as I am getting various requests for ASP.NET Webhook after my [event](http://www.indiamentor.com/events/coffee-tech-talks-gaurav-arora-asp-net-webhooks). I would be more than happy to see and be a part of contribution team of ASP.NET WebHook on ASP.NET Core.

**Comment by s-leonard** [comment]
Hi guys, do you have any updates on this or a rough estimate of when there will be ASP.NET Core support?

**Comment by Eilon** [comment]
@s-leonard we don't have an exact date, but we're working on getting a preview out that supports using web hooks receivers on ASP.NET Core. We'll update this issue when we know more.

**Comment by dougbu** [comment]
Core pieces checked in with 5f62336. Not closing this because too many receivers do not yet exist. And, checked-in code includes a fair number of TODO items and questions (`???`) but excludes tests.

**Comment by garora** [comment]
@dougbu - are you accepting `PR` for .NET Core version?

**Comment by dougbu** [comment]
@garora sure. But, please start with issues about the specific problem you'd like to address or gap you'd like to fill. Much better to triage issues than PRs. And, we'll avoid duplicating work.

**Comment by garora** [comment]
@dougbu - Cool, I would be starting soon. As we are managing two different flavor for webhooks one is for ASP.NET and another one is for ASP.NET Core so, do we have two separate branches or how you want to start to fix issues?

**Comment by dougbu** [comment]
Closing because remaining work is tracked in smaller issues. Though we'll reevaluate some design details and fill in gaps, the overall ASP.NET Core receiver basics and 10 receivers are in place.

**Comment by mkArtakMSFT** [comment]
Reopening this as a high-level tracking item for all the receivers work for 2.1 release

**Comment by dougbu** [comment]
What does "Working" mean here and why this issue is assigned to me? I have no action items I'm aware of that aren't already assigned to me.

We closed #5 when the port was complete and all known additional work for the 2.1 release (really 1.0 RTM) and beyond was tracked elsewhere.

**Comment by mkArtakMSFT** [comment]
Removed the `2 - Working` label as this is just an umbrella item, to track the rest of the work in this area for upcoming release.

**Comment by dougbu** [comment]
Added `cost - 0` since this is an umbrella tracking issue.

**Comment by dougbu** [comment]
Suggest we un-assign this issue and remove the `Mapped to Big Rock` label.

Alternatively, could close this issue as well as aspnet/Specs#103.

**Comment by dougbu** [comment]
@mkArtakMSFT @danroth27 @Eilon similar to aspnet/specs#103, I suggest we close this issue. Agreed?

**Comment by danroth27** [comment]
I think we can close this since implementing WebHooks support for ASP.NET Core is what this repo is all about.
